### PR TITLE
Address the CRAN issue, support macOS

### PR DESCRIPTION
**Pull request by kbenoit**
Attempting to:
* address the CRAN comments from #2372 
* resolve the macOS not finding TBB issue (#2367)

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -4289,19 +4289,12 @@ TBBLIBS=
...
-  { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-printf "%s\n" "yes" >&6; }
-else
-  if test ${SH_TBB_EXISTS} = 1 ; then
-    TBB_EXISTS=no
-    { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: no" >&5
-printf "%s\n" "no" >&6; }
-    { printf "%s\n" "$as_me:${as_lineno-$LINENO}: WARNING: Intel TBB not installed; install TBB devel package for parallel processing or update PKG_CONFIG_PATH environment variable" >&5
-printf "%s\n" "$as_me: WARNING: Intel TBB not installed; install TBB devel package for parallel processing or update PKG_CONFIG_PATH environment variable" >&2;}
+if pkg-config --exists tbb >/dev/null 2>&1; then
```
Comment (kbenoit) [inline configure:4292]: Addresses a) from #2372

File: configure [hunk 2]
```diff
@@ -4337,7 +4351,15 @@ int main() {
   }
 }
 _EOCONF
-  ${CXX} ${CXXFLAGS} ${TBBFLAGS} -o libtbb_test libtbb_test.cpp ${TBBLIBS} 2> errors.txt
+  
+  # Assuming TBBLIBS is already set from `pkg-config --libs tbb`
```
Comment (kbenoit) [inline configure:4355]: Addresses c)

File: configure [hunk 3]
```diff
@@ -4311,11 +4304,32 @@ printf "%s\n" "$as_me: WARNING: pkg-config not found; install pkg-config to auto
...
+#   else
+#     TBB_EXISTS=no
+#     { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: no" >&5
+# printf "%s\n" "no" >&6; }
+#     { printf "%s\n" "$as_me:${as_lineno-$LINENO}: WARNING: pkg-config not found; install pkg-config to auto-detect Intel TBB; assumed absent" >&5
+# printf "%s\n" "$as_me: WARNING: pkg-config not found; install pkg-config to auto-detect Intel TBB; assumed absent" >&2;}
+#   fi
+# fi
+
+if test "${TBB_EXISTS}" = yes ; then
```
Comment (kbenoit) [inline configure:4329]: This seems to solve #2367, so now, TBB is working on macOS!

**Comment by kbenoit** [comment]
@rsbivand @koheiw please check the code and see my comments and how they address what the CRAN team communicated. I still think we need to verify that we are answering b) - d) correctly.

**Comment by rsbivand** [comment]
@kbenoit Please never refer to `configure`, only ever `configure.ac`. Never edit `configure`. I don't see any of the above edits in `master`. I will never use `homebrew`, it is as bad as python wrt package maintenance discipline, and i9s not supported for use with R on macOS because of subtle differences in the build train. 

Here almost certainly homebrew has put `tbb.pc` somewhere else than the standard locations, of `/usr/lib*/pkgconfig` (on Linux, unsure on macOS), that is why the message says update PKG_CONFIG_PATH.

**Comment by rsbivand** [comment]
@kbenoit https://github.com/quanteda/quanteda/commit/839c648c55b937b03565758980191c56745b6751 should be abandoned. I have an M1, but will not risk TBB on that.

**Comment by rsbivand** [comment]
@kbenoit https://www.intel.com/content/www/us/en/developer/articles/system-requirements/intel-oneapi-base-toolkit-system-requirements.html

> NOTE Intel oneAPI Base Toolkit for macOS on x86 is discontinued in the 2024.0 release.  Several Intel-led open source developer tool projects will continue supporting macOS on Apple Silicon including Intel® oneAPI Threading Building Blocks (oneTBB) and Intel® Implicit SPMD Program Compiler and we welcome the opportunity to work with contributors to expand support to additional tools in the future.

So maybe Apple Silicon support (and macOS support more generally) isn't necessarily going to work. Perhaps oneTBB can be installed with the Intel installer, but it isn't obvious how.

**Comment by kbenoit** [comment]
> @kbenoit https://www.intel.com/content/www/us/en/developer/articles/system-requirements/intel-oneapi-base-toolkit-system-requirements.html
> 
> > NOTE Intel oneAPI Base Toolkit for macOS on x86 is discontinued in the 2024.0 release.  Several Intel-led open source developer tool projects will continue supporting macOS on Apple Silicon including Intel® oneAPI Threading Building Blocks (oneTBB) and Intel® Implicit SPMD Program Compiler and we welcome the opportunity to work with contributors to expand support to additional tools in the future.
> 
> So maybe Apple Silicon support (and macOS support more generally) isn't necessarily going to work. Perhaps oneTBB can be installed with the Intel installer, but it isn't obvious how, as [oneapi-src/oneTBB#806 (comment)](https://github.com/oneapi-src/oneTBB/issues/806#issuecomment-1378564856) suggests. The homebrew or coda installers are going to be subject to the risk of bitrot because they will stop at the last legacy Intel release which will be untried for newer hardware; all users should be using the latest version to patch for bugs and insecurities.

I understand the point, but most users will install it that way, plus the way we've relied on TBB for at least 6-7 years has been via RcppParallel, and the reason we had to migrate to using a separate installation in v4 was because of "bitrot" in that their current bundled version is from 2019.

Please see #2374 where I think I solved the macOS issue. I've now verified that it works on both x86 and silicon macOS.

**Comment by kbenoit** [comment]
> @kbenoit [839c648](https://github.com/quanteda/quanteda/commit/839c648c55b937b03565758980191c56745b6751) should be abandoned. I have an M1, but will not risk TBB on that.

I agree, I had misunderstood the configure.ac setup, all up to speed now though. I'll close this PR but leave the underlying issue open. I think I've resolved the mac issue in #2374.

**Pull request closed without merging**